connman/src/storage.c

```diff
@@ -794,11 +796,58 @@ static void keyfile_cleanup(void)
 	keyfile_hash = NULL;
 }
 
+static int validate_path(const char *pathname)
+{
+	char *expanded_path;
+	int err = 0;
+
+	/* realpath() expands all symbolic links and references. */
+	expanded_path = realpath(pathname, NULL);
+	if (!expanded_path) {
+		switch (errno) {
+		case EACCES:
+		case EINVAL: /* pathname was NULL */
+		case EIO:
+		case ELOOP:
+		case ENAMETOOLONG:
+		case ENOMEM:
+		case ENOTDIR:
+			DBG("realpath() error %d/%s", errno, strerror(errno));
+			return -errno;
+		/* When saving the file does not exist */
+		case ENOENT:
+			return 0;
+		}
+
+		return -errno;
+	}
+
+	/*
+	 * If the paths differ the path most likely contains symlink or is not
+	 * an absolute path.
+	 */
+	if (g_strcmp0(pathname, expanded_path))
+		err = -ENOTSUP;
+
+	g_free(expanded_path);
```

Comment:
expanded_path doesn't come from a GLib function, so normal free() should be used instead.

Reply:
Yeah, true. Changed and force pushed.

Reply:
Changed the approach so `g_free()` became valid after all :).